src/Host/Client/Impl/Host/BrokerClient.cs

```diff
@@ -87,10 +88,18 @@ internal abstract class BrokerClient : IBrokerClient {
         protected abstract void OnCredentialsValidated(bool isValid);
 
         public async Task PingAsync() {
-            try {
-                (await HttpClient.PostAsync("/ping", new StringContent(""))).EnsureSuccessStatusCode();
-            } catch (HttpRequestException ex) {
-                throw new RHostDisconnectedException(Resources.Error_HostNotResponding.FormatInvariant(ex.Message), ex);
+            if (HttpClient != null) {
+                var isOnline = await CheckMachineOnlineAsync();
```

Comment:
So putting ping it in ConnectAsync unfortunately gets this handled twice (main and intellisense sessions) and user gets two error messages


---
In reply to: [81589945](https://github.com/Microsoft/RTVS/pull/2416#discussion_r81589945) [](ancestors = 81589945,81588682,81588299,81587401,81587164)